guidance/_grammar.py

```diff
@@ -1,14 +1,18 @@
-import base64
-import uuid
-import json
-import inspect
-import types
 import re
+import types
 
-from typing import Any, Dict, List, TypeVar, Union
+from typing import Any, Dict, List, TYPE_CHECKING, TypeVar, Union
+
+try:
+    from . import _serialization_pb2
+    _serialization_pb2_is_imported = True
+except ImportError:
+    if TYPE_CHECKING:
+        raise
+    _serialization_pb2_is_imported = False
```

Comment:
This one may be one to leave out/force as a dependency. At some point, I think it'd be nice for ALL interfaces between models/engines to be passed through our serialization protocol, even local ones, so that we have a standardized contract.

Reply:
OK, I'll move `protobuf` back to required